### ISSUE TITLE
[BUG]: Kotlin runtime stackoverflow when trying to use String.replaceRange

### ISSUE THREAD

**Issue by partouf** [compiler-explorer/compiler-explorer#7016]
### Describe the bug

Found here https://social.vivaldi.net/@Patricia/113368474581710741

May or may not be related to stack size, might also be library `.jar` files that can't be loaded correctly at runtime

### Steps to reproduce

Stripped down version to reproduce: https://godbolt.org/z/j1ffKcfh3

### Expected behavior

No stackoverflow?

### Reproduction link

https://godbolt.org/z/j1ffKcfh3

### Screenshots

Not applicable

**Comment by partouf** [comment]
Might also be nice if we could somehow influence the jvm runtime arguments used here https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/compilers/kotlin.ts#L120 so users can experiment more. Maybe with an Override?

**Comment by junlarsen** [comment]
This happens when attempting to use anything from the Kotlin standard library

Also reproduces for https://godbolt.org/z/xnrjzjWxe

**Comment by junlarsen** [comment]
I don't think we link against kotlin-stdlib so a problem is expected but it's really weird that it causes a SO
